core/src/main/java/org/aerogear/mobile/core/http/pinning/CertificatePinningCheck.java

```diff
@@ -62,31 +71,31 @@ public void attachListener(@NonNull final CertificatePinningCheckListener listen
      * Remove the currently attached listener, if there is one attached.
      */
     public void detachListener() {
-        this.listener = null;
+        this.listener = DEFAULT_LISTENER;
     }
 
     /**
      * Perform a check and invoke a listener on response.
-     * 
+     *
      * @param url to which a request will be made
      */
     public void execute(@NonNull final String url) {
         HttpRequest request = this.httpModule.newRequest();
-        request.get(url);
-        HttpResponse httpResponse = request.execute();
+        request.get(url).respondWith(new Responder<HttpResponse>() {
+            @Override
+            public void onResult(HttpResponse httpResponse) {
+                CertificatePinningCheck.this.isSuccess = true;
+                if (CertificatePinningCheck.this.listener != null) {
```

Comment:
looks like the `listener` is always set, don't think this `if` check is required here.